### PR TITLE
Refactor tests

### PR DESCRIPTION
**Pull request by HeyNonster**
We've rewritten the tests to make it clearer what, exactly, they're testing and why they work the way that they do.

We renamed the database configuration keys from `test_host_1_db_1` to `test_pool_1_db_a`. Renaming `host` to `pool` feels better because each database configuration has the exact same host (`host: <%= mysql.host %>`)

We changed model names from things like `Test1` to `Pool1DbA` to make it clear that the model shares a connection pool with other models that start with `Pool1` and that it uses database  `A`.

This should help us work on making this gem `Rails 6.1` compatible because failing tests will be easier to diagnose without having to refer back to the model declarations to know which pool and database they refer to.

Co-authored-by: Cristina Matonte  <cristina.matonte@zendesk.com>

### CODE REVIEW

File: test/schema.rb [hunk 1]
```diff
@@ -8,8 +8,8 @@
 
   # Add a table only the shard database will have. Conditional
   # exists since Phenix loads the schema file for every database.
-  if ActiveRecord::Base.connection.current_database == 'arhp_test_1_shard'
-    create_table 'test1_shards' do |t|
+  if ActiveRecord::Base.connection.current_database == 'arhp_test_db_c'
+    create_table 'pool1_db_cs' do |t|
```
Comment (mriddle) [inline test/schema.rb:12]: Review tip: The `cs` here is due to us mimicking the pluralization functionality of Rails. E.g. shard model -> shards table

**Review by mriddle — APPROVED** [review]

**Review by anitsirc — APPROVED** [review]

File: test/database.yml [hunk 2]
```diff
@@ -1,76 +1,87 @@
...
   encoding: utf8
-  database: arhp_test_2
+  database: arhp_test_db_b
   username: <%= mysql.user %>
   password: <%= mysql.password %>
   host: <%= mysql.host %>
   reconnect: true
 
-test_host_2_db_3:
+
```
Comment (bquorning) [inline test/database.yml:38]:
```suggestion
```

File: test/test_arhp.rb [hunk 3]
```diff
@@ -30,34 +30,48 @@ def test_process_forking_with_connections
...
-    assert_equal(Test1.connection.raw_connection, Test2.connection.raw_connection)
-    assert_equal(Test3.connection.raw_connection, Test4.connection.raw_connection)
+  def test_models_with_matching_hosts_ports_sockets_usernames_and_replica_status_should_share_a_connection
+    assert_equal(Pool1DbA.connection.raw_connection, Pool1DbB.connection.raw_connection)
+    assert_equal(Pool2DbD.connection.raw_connection, Pool2DbE.connection.raw_connection)
   end
 
-  def test_models_without_matching_hosts_should_not_share_a_connection
-    refute_equal(Test1.connection.raw_connection, Test4.connection.raw_connection)
+  def test_models_without_matching_ports_should_not_share_a_connection
```
Comment (bquorning) [inline test/test_arhp.rb:38]:
```suggestion
  def test_models_with_different_ports_should_not_share_a_connection
```

File: test/test_arhp.rb [hunk 4]
```diff
@@ -30,34 +30,48 @@ def test_process_forking_with_connections
...
+    refute_equal(Pool1DbA.connection.raw_connection, Pool2DbD.connection.raw_connection)
   end
 
   def test_models_without_matching_usernames_should_not_share_a_connection
```
Comment (bquorning) [inline test/test_arhp.rb:42]:
```suggestion
  def test_models_with_different_usernames_should_not_share_a_connection
```

File: test/test_arhp.rb [hunk 5]
```diff
@@ -30,34 +30,48 @@ def test_process_forking_with_connections
...
-    refute_equal(Test4.connection.raw_connection, Test5.connection.raw_connection)
+    refute_equal(Pool2DbE.connection.raw_connection, Pool3DbE.connection.raw_connection)
   end
 
-  def test_models_without_match_slave_status_should_not_share_a_connection
-    refute_equal(Test1.connection.raw_connection, Test1Slave.connection.raw_connection)
+  def test_models_without_match_replica_status_should_not_share_a_connection
```
Comment (bquorning) [inline test/test_arhp.rb:46]:
```suggestion
  def test_models_with_different_replica_status_should_not_share_a_connection
```

**Review by bquorning — COMMENTED** [review]
Overall I really like these improvements, I think they make the tests much less confusing.

I have added some suggestions for changed wording. Feel free to reject them if you feel they don’t make things any clearer.

**Comment by HeyNonster** [comment]
> Overall I really like these improvements, I think they make the tests much less confusing.
> 
> I have added some suggestions for changed wording. Feel free to reject them if you feel they don’t make things any clearer.

I think all of your changes are perfect! Committed them and going to rebase our working Rails 6.1 branch onto them.

**Review by bquorning — APPROVED** [review]

**Review by bquorning — APPROVED** [review]

**Pull request merged**